python/lance/types.py

```diff
@@ -0,0 +1,122 @@
+#  Copyright 2022 Lance Authors
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""
+Arrow extension types for Lance
+"""
+from abc import ABC, abstractproperty
+import platform
+
+import pandas as pd
+
+import pyarrow as pa
+
+
+__all__ = ["ImageType", "ImageUriType", "ImageBinaryType", "Point2dType", "Box2dType"]
```

Comment:
Should we not expose ` ImageType`, which is the base class?

Reply:
ImageType.from_storage allows the user to get the correct subtype directly from storage array so they don't need to think about whether they need a uri or binary type